back/engines/commercial/user_custom_fields/app/serializers/user_custom_fields/web_api/v1/user_custom_field_serializer.rb

```diff
@@ -21,6 +21,15 @@ class UserCustomFieldSerializer < ::WebApi::V1::CustomFieldSerializer
         ) do |record, _params|
           record.current_ref_distribution
         end
+
+        has_many :projects, serializer: ::WebApi::V1::ProjectSerializer do |object|
+          project_ids = object.permissions.filter_map { |p| p.permission_scope_id if p.permission_scope_type == 'Project' }
+          phase_ids = object.permissions.filter_map { |p| p.permission_scope_id if p.permission_scope_type == 'Phase' }
+          if phase_ids.present?
+            project_ids += Phase.find(phase_ids).map(&:project_id)
```

Comment:
It's possible to use `.pluck(:project_id)` instead of `.map(&:project_id)`. The difference is that `pluck` will do only a select of the project_id, while `map` will force Rails to load all phases as instances into memory first. `pluck` is therefore a bit more efficient.